src/test/java/io/burt/athena/support/QueryExecutionHelper.java

```diff
@@ -143,28 +148,24 @@ public void queueGetQueryExecutionException(Exception e) {
         if (delay.isZero()) {
             return future;
         } else {
-            CompletableFuture<T> newFuture = new CompletableFuture<>();
-            ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
-            scheduler.schedule(
-                    () -> {
-                        try {
-                            newFuture.complete(future.get());
-                        } catch (ExecutionException e) {
-                            newFuture.completeExceptionally(e.getCause());
-                        } catch (Exception e) {
-                            newFuture.completeExceptionally(e);
-                        } finally {
-                            clock.tick(delay);
-                        }
-                    },
-                    delay.toMillis(),
-                    TimeUnit.MILLISECONDS
-            );
-            scheduler.shutdown();
-            return newFuture;
+            TestDelayedCompletableFuture<T> testFuture = new TestDelayedCompletableFuture<>(future, delay, clock);
+            CompletableFuture<T> restrictedFuture = mockCompletableFuture(invocation -> {
+                throw new UnsupportedOperationException(invocation.getMethod().toString());
+            });
+            try {
+                doAnswer(AdditionalAnswers.delegatesTo(testFuture)).when(restrictedFuture).get(anyLong(), any());
+            } catch (ExecutionException | InterruptedException | TimeoutException e) {
+                e.printStackTrace();
+            }
+            return restrictedFuture;
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private <T> CompletableFuture<T> mockCompletableFuture(Answer defaultAnswer) {
+        return mock(CompletableFuture.class, defaultAnswer);
+    }
```

Comment:
Why is this a method when it's just a line (plus maybe the supression annotation) and only called once?

Reply:
It's there precisely to have somewhere to put the unchecked annotation. I suppose it would be terrible to have it around the whole caller, but as far as I'm concerned it's standard practice to wrap these in methods.

Looking back at the code, I now think it would be more reasonable to also put the "protection" code in the test class and only allow "safe" construction of it. One benefit of moving it would be that the one-line method wouldn't feel quite as misplaced in that class as in this one.

Reply:
You can put the annotation on variable assignments too, so it should be possible to inline this.